ghcide/src/Development/IDE/LSP/Notifications.hs

```diff
@@ -47,15 +47,15 @@ setHandlersNotifications = PartialHandlers $ \WithMessage{..} x -> return x
             whenUriFile _uri $ \file -> do
                 -- We don't know if the file actually exists, or if the contents match those on disk
                 -- For example, vscode restores previously unsaved contents on open
-                modifyFilesOfInterest ide (M.insert file Modified)
+                modifyFilesOfInterest ide (M.insert file Modified{firstOpen=True})
                 setFileModified ide False file
                 logInfo (ideLogger ide) $ "Opened text document: " <> getUri _uri
 
     ,LSP.didChangeTextDocumentNotificationHandler = withNotification (LSP.didChangeTextDocumentNotificationHandler x) $
         \_ ide (DidChangeTextDocumentParams identifier@VersionedTextDocumentIdentifier{_uri} changes) -> do
             updatePositionMapping ide identifier changes
             whenUriFile _uri $ \file -> do
-              modifyFilesOfInterest ide (M.insert file Modified)
+              modifyFilesOfInterest ide (M.insert file Modified{firstOpen=False})
```

Comment:
Does this work when we rename a file?

Reply:
Why wouldn't it?

Reply:
Hm... maybe I missed something. It actually works.